relayer/relayer.go

```diff
@@ -83,6 +83,14 @@ type Response struct {
 type Protocol interface {
 	Endpoints(ServiceID) (map[AppAddr][]Endpoint, error)
 	SendRelay(Request) (Response, error)
+	ProtocolCache
+}
+
+// ProtocolCache reports if the protocol's cache is warm,
+// i.e. if it has the required data to serve requests.
+type ProtocolCache interface {
+	Name() string
```

Comment:
1. Why don't we just import/embed the HealthCheck interface in here (if that's the goal)
2. Alternatively, maybe this should be rename to `IsWarm` or `IsReady` instead of `IsAlive` (that's more of a health check thing)

Reply:
Not possible; it leads to an import cycle between `router` and `relayer`. In addition, the `router` package should not be imported elsewhere; the interface should be defined in the package it's used in. 

In order to solve the import cycle issue I've lifted the health check to its own `health` package and imported elsewhere. This solves both this issue and [the issue of not being able to define the interface check in the `shannon/morse` packages.](https://github.com/buildwithgrove/path/pull/20#discussion_r1763920313)